app/uk/gov/hmrc/cdsreimbursementclaimfrontend/controllers/claims/CheckClaimantDetailsController.scala

```diff
@@ -107,14 +106,17 @@ class CheckClaimantDetailsController @Inject() (
             formOk =>
               formOk match {
                 case YesClaimantDetailsAnswer =>
-                  Redirect(router.nextPageForChangeClaimantDetails(formOk))
+                  Redirect(router.nextPageForChangeClaimantDetails(formOk, featureSwitch))
                 case NoClaimantDetailsAnswer  =>
-                  val updatedClaim = removeContactDetails(fillingOutClaim)
+                  val draftC285Claim = fillingOutClaim.draftClaim.fold(identity)
+                  val updatedClaim   = fillingOutClaim.copy(draftClaim =
+                    draftC285Claim.copy(mrnContactDetailsAnswer = None, mrnContactAddressAnswer = None)
```

Comment:
Lines 111 and 112 can be replaces with FillingOutClaim.of(fillingOutClaim)(_.copy(mrnContactDetailsAnswer = None, mrnContactAddressAnswer = None))

Perhaps `of` can be renamed to something better like `copy` or `amend`. Feel free to change it.

Reply:
Done